apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Logging/LoggingTest.cs

```diff
@@ -0,0 +1,335 @@
+            DateTime startTime = DateTime.UtcNow;
+
+            var builder = new WebHostBuilder().UseStartup<SizedBufferLoggerTestApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                await client.GetAsync($"/Main/Warning/{testId}");
+                await client.GetAsync($"/Main/Error/{testId}");
+                await client.GetAsync($"/Main/Critical/{testId}");
+            }
+
+            // No entries should be found as not enough entries were created to
+            // flush the buffer.
+            Assert.Null(GetEntries(startTime, testId, 0));
+        }
+
+        [Fact]
+        public async Task Logging_NoBuffer()
+        {
+            string testId = Utils.GetTestId();
+            DateTime startTime = DateTime.UtcNow;
+
+            var builder = new WebHostBuilder().UseStartup<NoBufferLoggerTestApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                await client.GetAsync($"/Main/Debug/{testId}");
+                await client.GetAsync($"/Main/Info/{testId}");
+                await client.GetAsync($"/Main/Warning/{testId}");
+                await client.GetAsync($"/Main/Error/{testId}");
+                await client.GetAsync($"/Main/Critical/{testId}");
+            }
+
+            // NoBufferLoggerTestApplication does not support debug or info logs.
+            var results = GetEntries(startTime, testId, 3);
+            Assert.Equal(3, results.Count());
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Warning));
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Error));
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Critical));
+        }
+
+        [Fact]
+        public async Task Logging_SizedBuffer()
+        {
+            string testId = Utils.GetTestId();
+            DateTime startTime = DateTime.UtcNow;
+
+            var builder = new WebHostBuilder().UseStartup<SizedBufferLoggerTestApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                for (int i = 0; i < 250; i++)
+                {
+                    await client.GetAsync($"/Main/Debug/{testId}");
+                    await client.GetAsync($"/Main/Info/{testId}");
+                    await client.GetAsync($"/Main/Warning/{testId}");
+                    await client.GetAsync($"/Main/Error/{testId}");
+                    await client.GetAsync($"/Main/Critical/{testId}");
+                    await client.GetAsync($"/Main/Exception/{testId}");
+                }
+            }
+
+            // Just check that a large portion of logs entires were pushed.  Not all
+            // will be pushed as some may be in the buffer.
```

Comment:
No this shouldn't be flaky.  I just want to ensure the bulk of the logs are pushed.